### DISCUSSION TITLE
Spaceship operator and std

### DISCUSSION THREAD

**Discussion by kovarex** [microsoft/STL#1758] [General] [answered]
Hello, I'm now here, so I don't know how it works, I'm sorry if my questions are too obvious.

I would like to know what is the plan related to the support of the spaceship oerator.
I noticed, that this was https://github.com/microsoft/STL/pull/1206/files merged. When such a merge occurs, how can I know when/in which version it starts to be the part of the vs2019 install?

Also, what is the plan with the other classes like std::string for example. Is anyone planning/doing support for these, or should I try to write the code and make a pull request personally?

**Comment by StephanTLavavej** [comment, marked as answer]
Welcome! As you noticed with #1206, the spaceship operators were implemented gradually in a feature branch, `feature/spaceship`. We recently finished that work (including `string`, see #1635) and merged everything into `main`, see #1678. This will be available in VS 2019 16.10 Preview 2 - see https://github.com/microsoft/STL/wiki/Changelog where we list merged PRs and which VS releases they'll be available in.

**Reply by kovarex** [reply]
Thanks for the informative reply and the work on the stl. Looking forward to use it!
